ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java

```diff
@@ -108,59 +106,66 @@ private BlockAwareOperationTracer getBlockImportTracer(
+              .orElse(
+                  (__) -> {
+                    LOG.trace("Block Import uses NO_TRACING");
+                    return BlockAwareOperationTracer.NO_TRACING;
+                  });
     }
 
     return blockImportTracerProvider.getBlockImportTracer(header);
   }
 
+  /**
+   * Processes the block with no privateMetadata and no preprocessor.
+   *
+   * @param protocolContext the current context of the protocol
+   * @param blockchain the blockchain to append the block to
+   * @param worldState the world state to apply changes to
+   * @param block the block to process
+   * @return the block processing result
+   */
   @Override
   public BlockProcessingResult processBlock(
       final ProtocolContext protocolContext,
       final Blockchain blockchain,
       final MutableWorldState worldState,
-      final BlockHeader blockHeader,
-      final List<Transaction> transactions,
-      final List<BlockHeader> ommers,
-      final Optional<List<Withdrawal>> maybeWithdrawals,
-      final PrivateMetadataUpdater privateMetadataUpdater) {
+      final Block block) {
     return processBlock(
-        protocolContext,
-        blockchain,
-        worldState,
-        blockHeader,
-        transactions,
-        ommers,
-        maybeWithdrawals,
-        privateMetadataUpdater,
-        new NoPreprocessing());
+        protocolContext, blockchain, worldState, block, Optional.empty(), new NoPreprocessing());
   }
 
-  @VisibleForTesting
-  protected BlockProcessingResult processBlock(
+  @Override
+  public BlockProcessingResult processBlock(
       final ProtocolContext protocolContext,
       final Blockchain blockchain,
       final MutableWorldState worldState,
-      final BlockHeader blockHeader,
-      final List<Transaction> transactions,
-      final List<BlockHeader> ommers,
-      final Optional<List<Withdrawal>> maybeWithdrawals,
-      final PrivateMetadataUpdater privateMetadataUpdater,
+      final Block block,
+      final Optional<PrivateMetadataUpdater> privateMetadataUpdater,
       final PreprocessingFunction preprocessingBlockFunction) {
     final List<TransactionReceipt> receipts = new ArrayList<>();
     long currentGasUsed = 0;
     long currentBlobGasUsed = 0;
 
+    var blockHeader = block.getHeader();
+    var blockBody = block.getBody();
+    var ommers = blockBody.getOmmers();
+    var transactions = blockBody.getTransactions();
+    var maybeWithdrawals = blockBody.getWithdrawals();
+
     final ProtocolSpec protocolSpec = protocolSchedule.getByBlockHeader(blockHeader);
     final BlockHashLookup blockHashLookup =
         protocolSpec.getBlockHashProcessor().createBlockHashLookup(blockchain, blockHeader);
+
+    final BlockAwareOperationTracer blockTracer =
+        getBlockImportTracer(protocolContext, blockHeader);
+
+    LOG.trace("traceStartBlock for {}", blockHeader.getNumber());
+    blockTracer.traceStartBlock(blockHeader, blockHeader.getCoinbase());
```

Comment:
This should use the `blockHeader.getCoinbase()`. Instead, it should use the `miningBeneficiary` calculated below

Reply:
![image](https://github.com/user-attachments/assets/fdaaee9f-211f-40b4-b302-2867b65cf0f7)
Thanks for finding this!